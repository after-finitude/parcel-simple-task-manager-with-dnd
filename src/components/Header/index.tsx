import React from "react";
import Typography from "@material-ui/core/Typography";

export const Header: React.FC = () => (
  <Typography variant="h3">Task Manager</Typography>
);
