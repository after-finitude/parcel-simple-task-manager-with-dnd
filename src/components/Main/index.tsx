import React from "react";
import { useSelector, shallowEqual } from "react-redux";
import styled from "styled-components";

import { RootState } from "store/rootReducer";
import { DialogState, DialogTypes } from "store/dialog/types";

import { TasksList } from "components/Tasks/TasksList";
import { TasksAddButton } from "components/Tasks/TasksAddButton";
import { TasksCreateDialog } from "components/Tasks/TasksCreateDialog";
import { TasksUpdateDialog } from "components/Tasks/TasksUpdateDialog";

export const Main: React.FC = () => {
  const dialog = useSelector<RootState, DialogState>(
    state => state.dialog,
    shallowEqual,
  );

  return (
    <MainLayout>
      <TasksList />
      <TasksAddButton />
      {dialog.type === DialogTypes.CREATE ? (
        <TasksCreateDialog status={dialog.status} />
      ) : (
        <TasksUpdateDialog
          status={dialog.status}
          currentTaskId={dialog.currentTaskId}
        />
      )}
    </MainLayout>
  );
};

const MainLayout = styled.div`
  width: 25%;
`;
