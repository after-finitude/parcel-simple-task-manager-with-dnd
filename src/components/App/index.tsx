import React from "react";
import styled from "styled-components";

import { Header } from "components/Header";
import { Main } from "components/Main";

export const App: React.FC = () => {
  return (
    <BaseLayout>
      <Header />
      <Main />
    </BaseLayout>
  );
};

const BaseLayout = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;
