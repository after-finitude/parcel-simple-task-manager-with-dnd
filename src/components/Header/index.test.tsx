import React from "react";
import { render } from "enzyme";

import { Header } from "components/Header";

describe("Header", () => {
  it("should render correctly", () => {
    const component = render(<Header />);

    expect(component).toMatchSnapshot();
  });
});
