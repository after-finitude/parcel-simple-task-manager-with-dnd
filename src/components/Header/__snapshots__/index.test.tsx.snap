// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<h3
  class="MuiTypography-root MuiTypography-h3"
>
  Task Manager
</h3>
`;
